src/NUnitFramework/tests/Attributes/ValueSourceTests.cs

```diff
@@ -233,7 +233,8 @@ public void MethodWithArrayArguments([ValueSource(nameof(ComplexArrayBasedTestIn
             new object[] { 1, "text", new object() },
             new object[0],
             new object[] { 1, new int[] { 2, 3 }, 4 },
-            new object[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 }
+            new object[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 },
+            new object[] { new byte[,] { { 1, 2 }, { 2, 3 } } }
```

Comment:
@mikkelbu Since NUnit is inconsistent in how it handles splatting, and for better reading comprehension, could you put them on the same level?
```cs
new object[] { new object[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 } },
new object[] { new byte[,] { { 1, 2 }, { 2, 3 } } }
```

Reply:
@CharliePoole Yes. This is probably going to tie into an existing PR and several other issues though, addressing consistency between TestCaseSource and ValueSource. Splatting means applying an array or tuple so that its elements become arguments to some method call.

Reply:
Yes, it does. 😄

At the level where we are working, there's always an explicit object array that holds the args, but not necessarily at the user level. Key issue is how we interpret a __user-specified__ array, particularly when passed to a method that accepts a single object arg. Is the outer object array just a holder for the arg or is it the arg itself?

That said, this is not the issue here. We are trying to figure out how to represent a __single__ argument.

In any case, I'd love to see an issue that highlights any inconsistency you have found, especially in the key case of an object array passed to a method taking a single object. Our intent, obviously, has always been to be consistent but since the code is necessarily in multiple places, it has a tendency to degrade.

Reply:
I was confused. TestCaseSource enumerates argument lists; ValueSource enumerates arguments.

What's weird is that TestCaseSource passes the *argument list* `new object[] { 1, 2, 3 }` as the *argument* `new object[] { 1, 2, 3 }` rather than giving the error that there are not three parameters in the test method.

If it gave an error, that would make it symmetric with ValueSource and consistent when wrapped with TestCaseData.

<details>

```cs
using System.Collections.Generic;
using NUnit.Framework;

public static class Repro
{
    public static readonly IEnumerable<object[]> ArrayPerArgumentList = new[]
    {
        new object[] { 1, 2, 3 },
        new object[] { new object[] { 1, 2, 3 } }
    };

    [TestCaseSource(nameof(ArrayPerArgumentList))]
    public static void TestA(object anythingGoes)
    {
        // WithTestCaseSource([1, 2, 3]) ✔ (❕)
        // WithTestCaseSource([1, 2, 3]) ✔
    }

    public static readonly IEnumerable<TestCaseData> TestCaseDataPerArgumentList = new[]
    {
        new TestCaseData(new object[] { 1, 2, 3 }),
        new TestCaseData(new object[] { new object[] { 1, 2, 3 } })
    };

    [TestCaseSource(nameof(TestCaseDataPerArgumentList))]
    public static void TestB(object anythingGoes)
    {
        // WithTestCaseSource(1,2,3)     ❌ Too many arguments provided, provide at most 1 arguments.
        // WithTestCaseSource([1, 2, 3]) ✔
    }
}
```

</details>

But I'm not sure whether this is a good or bad thing.

Reply:
So when you use TestCaseSource with an object[], NUnit has to figure out what you meant.

If you pass `new object[] { 1, 2, 3 }` to a method taking a single int, it should create two test cases. But if you pass the same thing to a method taking a single object or object[], then that's just one test case. If you want it to be three test cases, then you would have to use `TestCaseData` instead of `object[]`. `TestCaseData` always makes it unambiguous.

Probably, as a separate issue, we should review all the things we allow the user to pass in as data and how we handle it. We were probably too generous in what we originally accepted, although I'm not sure if we can get away from it now other than through documentation.

Ahha! I just opened up your hidden example, and I see I'm telling you what you already know!